lib/SIL/SILPrinter.cpp

```diff
@@ -237,10 +237,13 @@ static void printValueDecl(ValueDecl *Decl, raw_ostream &OS) {
   printFullContext(Decl->getDeclContext(), OS);
   assert(Decl->hasName());
 
-  if (Decl->isOperator())
+  if (Decl->isOperator()) {
     OS << '"' << Decl->getBaseName() << '"';
-  else
+  } else if (Decl->getBaseName() == Decl->getASTContext().Id_subscript) {
```

Comment:
I feel like you should drop "subscript" from KnownIdentifiers.def at this point.

(Also I still think escaping all keywords is the way to go in the future, but it's fine for now.)